src/removeHighlightWrappers.ts

```diff
@@ -12,52 +12,59 @@ const NODE_TYPE = {
  * Removes highlights from element. If element is a highlight itself, it is removed as well.
  * If no element is given, all highlights all removed.
  * @param {HTMLElement} [element] - element to remove highlights from
+ * @param {HTMLElement} [filterFn] - filter which highlights should be removed from the element, default: all
  */
-export function removeAllHighlights(element: HTMLElement) {
-  getHighlights(element).forEach(removeHighlightElement);
+export function removeAllHighlights(element: HTMLElement, filterFn?: (element: HTMLElement) => boolean) {
+  getHighlights(element).forEach((element) => removeHighlightElement(element, filterFn));
 }
 
-export default function(highlight: Highlight) {
-  highlight.elements.forEach(removeHighlightElement);
+export default function(highlight: Highlight, filterFn?: (element: HTMLElement) => boolean) {
+  highlight.elements.forEach((element) => removeHighlightElement(element, filterFn));
 }
 
-function removeHighlightElement(element: HTMLElement) {
+function removeHighlightElement(element: HTMLElement, filterFn?: (element: HTMLElement) => boolean) {
   const container = element,
     highlights = getHighlights(container);
 
-  function mergeSiblingTextNodes(textNode: Text) {
-    const prev = textNode.previousSibling,
-      next = textNode.nextSibling;
+  const filteredHighlights = filterFn
+    ? highlights.filter(filterFn)
+    : highlights;
+
+  function mergeSiblingNodes(node: Node) {
+    if (node.nodeType !== NODE_TYPE.TEXT_NODE) { return; }
```

Comment:
was this just broken before? or did there used to be something else filtering the node type?